### ISSUE TITLE
Iterate through `DataLoader` create from dataset of stream

### ISSUE THREAD

**Issue by tungts1101** [ContinualAI/avalanche#1666]
🐛 **Describe the bug**
Unclear way to iterate through pytorch `DataLoader`

🐜 **To Reproduce**
```
import torch
from torch import nn
from torch.nn import functional as F
from torch import optim
from torch.utils.data import DataLoader
from torchvision import models, transforms

from avalanche.benchmarks.classic import SplitCIFAR10
from avalanche.evaluation.metrics import accuracy_metrics, forgetting_metrics, loss_metrics
from avalanche.logging import InteractiveLogger, TextLogger
from avalanche.training.storage_policy import ReservoirSamplingBuffer
from avalanche.training.plugins import EvaluationPlugin
from avalanche.benchmarks.utils import as_classification_dataset

train_transform = transforms.Compose([
    transforms.RandomCrop(32, padding=4),
    transforms.RandomHorizontalFlip(),
    transforms.ToTensor(),
    transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
])
test_transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
])
benchmark = SplitCIFAR10(
    seed=seed,
    n_experiences=5, 
    return_task_id=False, 
    train_transform=train_transform,
    eval_transform=test_transform)

for experience in benchmark.train_stream:
    print(f"Starting training on experience {experience.current_experience}")
    print(f"Number of samples in this experience: {len(experience.dataset)}")
    train_loader = DataLoader(experience.dataset, batch_size=batch_size, shuffle=True)
    
    model.train()
    for batch_idx, (x, y) in enumerate(train_loader):
        pass
```

🐝 **Expected behavior**
No error is showing up.

🐞 **Screenshots**
![image](https://github.com/user-attachments/assets/d952c748-4d99-42cf-b560-c1987fbfa45e)

🦋 **Additional context**
I am learning how to use Avalanche by implementing the strategy for DER++. At the moment I want to iterate through the dataset obtained from the stream but I keep getting errors as shown in the image above. I would appreciate if there is a place showing an example of using my own model with benchmark and evaluation of Avalanche.

**Comment by AntonioCarta** [comment]
samples are triplets `<inputs, targets, task_labels>`. Task labels are all set to 0 if the benchmark is not multi-task.
